contracts/contracts/TACoApplication.sol

```diff
@@ -428,18 +450,21 @@ contract TACoApplication is
     }
 
     /**
-     * @notice Withdraw available amount of T reward to beneficiary. Can be called only by beneficiary
+     * @notice Withdraw available amount of T reward to beneficiary. Can be called only by beneficiary or by reward distribution contract
      * @param _stakingProvider Staking provider address
      */
     function withdrawRewards(address _stakingProvider) external updateReward(_stakingProvider) {
         address beneficiary = getBeneficiary(_stakingProvider);
-        require(msg.sender == beneficiary, "Caller must be beneficiary");
+        require(
```

Comment:
Out of curiosity - what's the concern if any random entity (not beneficiary or reward contract) calls this function?

Reply:
I can't remember what was exactly underneath, my guess is in case it's tax related operation then someone can troll with it

Reply:
Ah right - that makes sense 👍 .

Reply:
I guess to that point - presumably there are restrictions on who can call the corresponding reward contract method (which then calls this method)?